PWGHF/TableProducer/trackIndexSkimCreator.cxx

```diff
@@ -3222,18 +3227,85 @@ struct HfTrackIndexSkimCreatorLfCascades {
           }
 
           // check not to take the same particle twice in the decay chain
-          if (trackPion1.globalIndex() == trackXiDauCharged.globalIndex() || trackPion1.globalIndex() == trackV0DauPos.globalIndex() || trackPion1.globalIndex() == trackV0DauNeg.globalIndex() || trackPion1.globalIndex() == casc.globalIndex()) {
+          if (trackPion1.globalIndex() == trackXiDauCharged.globalIndex() || trackPion1.globalIndex() == trackV0PosDau.globalIndex() || trackPion1.globalIndex() == trackV0NegDau.globalIndex()) {
             continue;
           }
 
           // primary pion track to be processed with DCAFitter
           auto trackParVarPion1 = getTrackParCov(trackPion1);
 
+          // find charm baryon decay using xi PID hypothesis
+          int nVtxFrom2ProngFitterXiHyp = df2.process(trackCascXi, trackParVarPion1);
+          if (nVtxFrom2ProngFitterXiHyp > 0) {
+
+            df2.propagateTracksToVertex();
+
+            std::array<float, 3> pVecXi = {0.};
+            std::array<float, 3> pVecPion1XiHyp = {0.};
+            df2.getTrack(0).getPxPyPzGlo(pVecXi);
+            df2.getTrack(1).getPxPyPzGlo(pVecPion1iHyp);
+
+            auto secondaryVertex2XiHyp = df2.getPCACandidate();
+
+            cascType = 0;
+            SETBIT(cascType, kXi);
+
+            // fill table row
+            rowTrackIndexCasc2Prong(thisCollId,
+                                    casc.globalIndex(),
+                                    trackPion1.globalIndex(),
+                                    cascType);
```

Comment:
I would use `HFflag` here. Also, you should not fill it until it is not checked also the other mass hypothesis. Moreover, it must be filled only once at the end and `HFflag` will contain the information whether it is compatible with Xi, Omega, or both.